tests/test_discounts.py

```diff
@@ -201,6 +202,52 @@ def test_invalid_checkout_discount_form(monkeypatch, voucher):
     assert 'voucher' in form.errors
 
 
+def test_voucher_queryset_active(voucher):
+    vouchers = Voucher.objects.all()
+    assert len(vouchers) == 1
+    active_vouchers = Voucher.objects.active(
+        date=date.today() - timedelta(days=1))
+    assert len(active_vouchers) == 0
+
+
+def test_checkout_discount_form_active_queryset_voucher_not_active(voucher):
+    assert len(Voucher.objects.all()) == 1
+    checkout = Mock(cart=Mock())
+    voucher.start_date = date.today() + timedelta(days=1)
+    voucher.save()
+    form = CheckoutDiscountForm({'voucher': voucher.code}, checkout=checkout)
+    qs = form.fields['voucher'].queryset
+    assert len(qs) == 0
+
+
+def test_checkout_discount_form_active_queryset_voucher_active(voucher):
+    assert len(Voucher.objects.all()) == 1
+    checkout = Mock(cart=Mock())
+    voucher.start_date = date.today()
+    voucher.save()
+    form = CheckoutDiscountForm({'voucher': voucher.code}, checkout=checkout)
+    qs = form.fields['voucher'].queryset
+    assert len(qs) == 1
+
+
+def test_checkout_discount_form_active_queryset_after_some_time(voucher):
+    with freeze_time(datetime.now()) as frozen_date:
```

Comment:
Why pass the current time to `freeze_time`? I was hoping for a test that goes like this:

* Set voucher's  start time to June 1 2016
* With freeze_time(2016, 5, 31) check that it does not work
* With freeze_time(2016, 6, 1) check that it does work

It's better to explicitly freeze the time around relevant code parts instead of treating it like a global mock.

Reply:
The problem was that evaluating active vouchers was done only once with 'server start'. I am not sure if using freeze_time with defined time will recreate this behavior in pytest.
But of course i can change that.

Reply:
The idea is to use two different points in time to prove that its behavior changes.